pallets/settlement/src/lib.rs

```diff
@@ -565,6 +568,9 @@ decl_storage! {
         /// Venues that are allowed to create instructions involving a particular ticker. Oly used if filtering is enabled.
         /// (ticker, venue_id) -> allowed
         VenueAllowList get(fn venue_allow_list): double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) u64 => bool;
+        /// To know whether the instruction is scheduled or not.
+        /// (instruction_id) => boolean value to know whether instruction is scheduled or not.
+        ScheduledInstructions get(fn has_instruction_already_scheduled): map hasher(identity) u64 => bool;
```

Comment:
We don't need this at all. Since schedule names are deterministic, we can simply do a lookup in the scheduler pallet or just call the schedule function and handle the error.

Reply:
There is no way to lookup the scheduler pallet, That is the reason I used this.

Reply:
> just call the schedule function and handle the error.

Reply:
There is a downside, in this case, is when we only want to do operation on the basis that instruction is already scheduled and skip when it is not scheduled.

Reply:
I can only see `cancel_named` and `schedule_named`. Both can be called without checking this variable and we can handle/ignore the errors.

Reply:
Yeah but the fun part is cancel_named does not emit any error so no way to handle the error.

Reply:
You don't need to handle any errors in `cancel_named`. Either case is fine with us. We just want the schedule to not exist. It does not matter if it gets canceled or if it was not present at all.

Reply:
Then I think we don't need to take care of the error in the `schedule_named` as well because if an error happens then we always assume that it exists if don't then proceed because `schedul_named` does also not emit an error.